kernel/sched.c

```diff
@@ -493,11 +493,11 @@ void k_sched_time_slice_set(int32_t slice, int prio)
 }
 
 #ifdef CONFIG_TIMESLICE_PER_THREAD
-void k_thread_time_slice_set(struct k_thread *th, int32_t slice_ticks,
+void k_thread_time_slice_set(struct k_thread *th, int32_t thread_slice_ticks,
```

Comment:
This is a good example here: we happen to have a static variable in the file named "slice_ticks" which this parameter is shadowing.   But the bug isn't that this function picked a bad name for its parameter (it seems fine!) it's that our global namespace has a distressingly ambiguous name in it.  We could call it "thread_time_slice_ticks" or something, which IMHO would be better.  Even better would be to move it to the _kernel struct, which is probably where it should have been to begin with (the same is true for lots of statics in the kernel files.

Reply:
Hm... most of the changes in this PR are small enough that should be easy to do. Let me try that.

Reply:
I have changed some but not all, depending on whether the changes in local functions would make more sense, and also how invasive the changes would be. I haven't touched the time slicing one (mentioned above) as this would require changing other `slice_*` global variables... which may cause more problems. So I'll leave it as-is and will tackle this later.